packages/core/src/network/heartbeat.ts

```diff
@@ -140,6 +166,47 @@ export default class Heartbeat {
     }
   }
 
+  /**
+   * Recalculates the network health indicator based on the
+   * current network state knowledge.
+   * @returns Value of the current network health indicator (possibly updated).
+   */
+  public recalculateNetworkHealth(): NetworkHealthIndicator {
+    let newHealthValue = NetworkHealthIndicator.RED
+    let lowQualityPublic = 0,
+      lowQualityNonPublic = 0
+    let highQualityPublic = 0,
+      highQualityNonPublic = 0
```

Comment:
Despite it looks fancy, I recommend to use the same syntax everywhere, namely instantiate each variable seperately with `let` or `const`